app/src/main/java/com/bakkenbaeck/sol/viewModel/SunViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.bakkenbaeck.sol.viewModel
+
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.content.Intent
+import android.text.Spanned
+import com.bakkenbaeck.sol.BaseApplication
+import com.bakkenbaeck.sol.extension.calculateProgress
+import com.bakkenbaeck.sol.extension.toHtml
+import com.bakkenbaeck.sol.model.local.Phase
+import java.util.Calendar
+import java.util.Date
+
+class SunViewModel : ViewModel() {
+
+    val todaysMessage by lazy { MutableLiveData<Spanned>() }
+    val locationMessage by lazy { MutableLiveData<String>() }
+    val currentPhase by lazy { MutableLiveData<Phase>() }
+    val sunrise by lazy { MutableLiveData<Long>() }
+    val sunset by lazy { MutableLiveData<Long>() }
+    val progress by lazy { MutableLiveData<Double>() }
+    val date by lazy { MutableLiveData<Date>() }
```

Comment:
Don't expose `MutableLiveData`s from the ViewModel - expose them as `LiveData`s

Reply:
Could you elaborate please? If you could give me an example as well that would be great :)

Reply:
`MutableLiveData` has the ability to have its value set, `LiveData` does not.
So if you expose a `MutableLiveData` to the View layer, there is nothing to stop the View layer from setting the value.

There are a couple of ways to solve this outlined here: https://stackoverflow.com/a/45549765